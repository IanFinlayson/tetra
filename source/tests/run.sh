#!/bin/bash

echo "Running test suite..."

pass=0
fail=0
for file in *.ttr
do 
  ../../bin/tetra $file > ./output/tmp.txt
  diffWords=`diff ./output/tmp.txt ./output/${file}.out | wc -w`
  if [ $diffWords -ne 0 ]
  then
    fail=$(($fail + 1))
    echo "********** $file failed **********"
  else
    pass=$(($pass + 1))
  fi
done

echo "${pass}/$(($pass + $fail)) tests were successful."
