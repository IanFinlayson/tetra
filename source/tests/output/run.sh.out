The following error was detected in your program:
Error: (line 3) syntax error, unexpected IDENTIFIER

Execution aborted
